flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_Engine.js

```diff
@@ -266,23 +213,75 @@ var A320_Neo_LowerECAM_Engine;
                 }
             }
         }
-        setN1VibrationValue(_value, _force = false) {
-            if ((_value != this.N1VibrationValue) || _force) {
-                this.N1VibrationValue = _value;
+        setN1VibrationValue(value, force = false) {
+            if ((value != this.N1VibrationValue) || force) {
+                this.N1VibrationValue = value;
                 if (this.N1VibrationValueText != null) {
                     this.N1VibrationValueText.textContent = fastToFixed(this.N1VibrationValue, 1);
                 }
             }
         }
-        setN2VibrationValue(_value, _force = false) {
-            if ((_value != this.N2VibrationValue) || _force) {
-                this.N2VibrationValue = _value;
+        setN2VibrationValue(value, force = false) {
+            if ((value != this.N2VibrationValue) || force) {
+                this.N2VibrationValue = value;
                 if (this.N2VibrationValueText != null) {
                     this.N2VibrationValueText.textContent = fastToFixed(this.N2VibrationValue, 1);
                 }
             }
         }
     }
     A320_Neo_LowerECAM_Engine.EngineInfo = EngineInfo;
+    class EngineIgniter {
+        constructor(textElement) {
+            this.currentState = Definitions.IGN_STATE.NONE;
+            this._textElement = textElement;
+            this._lastUsedIgniter = Definitions.IGN_STATE.NONE;
+            this._isIgniting = false;
+        }
+        updateStatus(engineStarting, n2Igniting, n2Percent) {
+            const targetState = this.getTargetState(engineStarting, n2Igniting, n2Percent);
+            if (this.currentState !== Definitions.IGN_STATE.NONE && targetState !== Definitions.IGN_STATE.NONE) {
+                this._lastUsedIgniter = this.currentState;
+                this._isIgniting = true;
+            } else {
+                if (targetState !== this.currentState) {
+                    if (targetState === Definitions.IGN_STATE.NONE) {
+                        this._isIgniting = false;
+                    }
+                    this.currentState = targetState;
+                    if (this._textElement !== null) {
+                        this._textElement.textContent = this.getIGNStringFromState(targetState);
+                    }
+                }
+            }
+        }
```

Comment:
**issue**: `targetState` is only set when neither current nor target state are NONE.
**suggestion**: It seems to me (but I'm not sure) that the else should be removed. Which would result in the code suggestion below (I also removed the `_isIgniting` here, based on my other comment).
**suggestion**: Consider removing the `this._textElement !== null` check, as I don't think it will ever be null.

```suggestion
        updateStatus(engineStarting, n2Igniting, n2Percent) {
            const targetState = this.getTargetState(engineStarting, n2Igniting, n2Percent);
            if (this.currentState !== Definitions.IGN_STATE.NONE && targetState !== Definitions.IGN_STATE.NONE) {
                this._lastUsedIgniter = this.currentState;
            }
            
            if (targetState !== this.currentState && this._textElement !== null) {
                this._textElement.textContent = this.getIGNStringFromState(targetState);
            }
            
            this.currentState = targetState;
        }
```

Reply:
This on its own caused an issue where igniters alternated between A and B constantly. This was because getTargetState had no concept of currentState which led it to always set the target state as the opposite igniter. The old code prevented this from changing the actual igniter text but the target state did change.

That behaviour wasn't great, so the code now returns the currentState as targetState if the currentState is not NONE, as in it is running. We don't want to change our target state until it is stopped.